apps/api/src/domain/models/__tests__/utilities/buildAllInvalidTestCasesForResource.ts

```diff
@@ -0,0 +1,36 @@
+import DisallowedContextTypeForResourceError from '../../../domainModelValidators/errors/context/invalidContextStateErrors/DisallowedContextTypeForResourceError';
+import getValidEntityInstaceForTest from '../../../domainModelValidators/__tests__/domainModelValidators/utilities/getValidEntityInstaceForTest';
+import { ResourceType, ResourceTypeToInstance } from '../../../types/resourceTypes';
+import isContextAllowedForGivenResourceType from '../../allowedContexts/isContextAllowedForGivenResourceType';
+import { EdgeConnectionContextType } from '../../context/types/EdgeConnectionContextType';
+import { ResourceModelContextStateValidatorInvalidTestCase } from '../resourceModelContextStateValidators.spec';
+import getValidContextOfTypeForTest from './getValidContextOfTypeForTest';
+
+export default <TResourceType extends ResourceType = ResourceType>(
+    resourceType: TResourceType
+): ResourceModelContextStateValidatorInvalidTestCase<ResourceTypeToInstance[TResourceType]>[] => {
+    const resource = getValidEntityInstaceForTest(resourceType);
+
+    const result = Object.values(EdgeConnectionContextType)
+        .filter((contextType) => !isContextAllowedForGivenResourceType(contextType, resourceType))
+        .map((contextType) => ({
+            contextType,
+        }))
+        .map(({ contextType }) => {
+            const expectedError = new DisallowedContextTypeForResourceError(
+                contextType,
+                resource.getCompositeIdentifier()
+            );
+
+            const testCase = {
+                description: `when a ${resourceType} receives a context with disallowed type" ${contextType}`,
```

Comment:
Extra `"` in the message.  Should we stick to colons everywhere there's a dynamic var like `${resourceType]`?

Reply:
This is not an error message, it's a test description, so I'd prefer not to have colons here.

Reply:
However, there is a more important point I missed. I really should be using a formatter (view model > presentation utils) for resourceType and contextType, which are camel case internally. I'll add a tech-debt story for that.

Reply:
BTW I fixed the ", which I did intend to be a :